Tools/android/GenerateAndroidMk.py

```diff
@@ -0,0 +1,224 @@
+        #windows can help us debug the script
+        #but we do not want generate makefile on widnows
+        if os.name == "nt":
+            return
+        cmd = "rm " + path.join(self.build_, "CMakeCache.txt") + "> /dev/null 2>&1;"
+        cmd += "mkdir " + self.build_ + "> /dev/null 2>&1;"
+        cmd += "cd " + self.build_
+        cmd += "&& " + self.getCmakeCmd() + "> /dev/null 2>&1"
+        os.system(cmd)
+
+    def getIncludes(self):
+        text = self.getDefines("CXX_INCLUDES")
+        includes = []
+        lines = text.split("\n")
+        for l in lines:
+            #normpath will make sure we did not refer outside.
+            p = path.normpath(l)
+            j = p.find(self.src_)
+            if j != -1:
+                includes.append(p[j:].replace(self.src_, "$(LOCAL_PATH)"))
+        return INDENT + ("\n" + INDENT).join(includes)
+
+    def getDefines(self, name):
+        flags = self.getFlagsfile()
+        with open(flags) as f:
+            text = f.read()
+        line = re.findall(name + " =.*\n", text)[0]
+        line = line.replace(name + " = ", "")
+        line = INDENT + line.replace(" ", " \\\n" + INDENT)
+        return line
+
+    def getSources(self):
+        makefile = self.getMakefile()
+        with open(makefile) as f:
+            text = f.read()
+        lines = re.findall(".*?\\.o:\\n", text)
+        lines = [l.replace(".o:\n", " \\") for l in lines]
+        self.adjustSources(lines)
+        #make source pretty
+        return INDENT + ("\n" + INDENT).join(lines)
+
+
+class GmmGeneator(Generator):
+    def __init__(self, root):
+        src = path.join(root, "gmmlib")
+        templ = path.join(src,  "Tools/android")
+        super(GmmGeneator, self).__init__(src, templ)
+
+    def getTemplatePath(self):
+        return path.join(self.templ_, "gmm.Android.mk.template")
+
+    def getMakefile(self):
+        return path.join(self.build_, "Source/GmmLib/Makefile")
+
+    def getFlagsfile(self):
+        return path.join(self.build_, "Source/GmmLib/CMakeFiles/gmm_umd.dir/flags.make")
+
+    def adjustSources(self, lines):
+        for i, l in enumerate(lines):
+            j = l.find("__/")
+            if j == -1:
+                lines[i] = path.join("Source/GmmLib", l)
+            else:
+                lines[i] = path.join("Source", l[j+3:])
+
+
+class CmrtGeneator(Generator):
+    def __init__(self, root):
+        src = path.join(root, "media-driver/cmrtlib")
+        templ = path.join(src,  "android")
+        super(CmrtGeneator, self).__init__(src, templ)
+
+    def getTemplatePath(self):
+        return path.join(self.templ_, "cmrt.Android.mk.template")
+
+    def getMakefile(self):
+        return path.join(self.build_, "linux/Makefile")
+
+    def getFlagsfile(self):
+        return path.join(self.build_, "linux/CMakeFiles/igfxcmrt.dir/flags.make")
+
+    def adjustSources(self, lines):
+        for i, l in enumerate(lines):
+            j = l.find("__/")
```

Comment:
After you generated makefile from cmake, if you use `grep -ir \\.o:$ Makefile  |sed s/.c.o:/.c/ |sed s/.cpp.o:/.cpp/ | sed s,^,\\t, | sed s/$/' \\'/` to grep makefile in gmmlib, you will see __/xxxx, could we conside adjust the cmake to remove __/?